docs/Implementing-easyblocks.rst

```diff
@@ -0,0 +1,486 @@
+successful or not.
+
+This is done by redefining the ``sanity_check_step`` method in the easyblock. For example:
+
+.. code:: python
+
+    from easybuild.framework.easyblock import EasyBlock
+
+    class EB_Example(EasyBlock):
+        """Custom easyblock for Example"""
+
+        def sanity_check_step(self):
+            """Custom sanity check for Example."""
+
+            custom_paths = {
+                'files': ['bin/example'],
+                'dirs': [],
+            }
+            custom_commands = ['example --version']
+            super(EB_Example, self).sanity_check_step(custom_paths=custom_paths, custom_commands=custom_commands)
+
+
+You can both specify file path and subdirectories to check for, which are specified relative to the installation directory,
+as well as simple commands that should execute successfully after completing the installation and loading the generated module file.
+
+Note that it is up to you have specific you make the sanity check, but it is recommended to make the check as complete
+as possible to catch any potential build or installation problems thay may occur.
+
+
+.. _implementing_easyblocks_version_specific:
+
+Version-specific parts
+~~~~~~~~~~~~~~~~~~~~~~
+
+In some case, version-specific actions or checks need to be included in an easyblock.
+For this, it is recommended to use ``LooseVersion`` rather than directly comparing version numbers using string values.
+
+For example:
+
+
+.. code:: python
+
+    from distutils.version import LooseVersion
+
+    from easybuild.framework.easyblock import EasyBlock
+
+    class EB_Example(EasyBlock):
+        """Custom easyblock for Example"""
+
+        def sanity_check_step(self):
+            """Custom sanity check for Example."""
+
+            custom_paths = {
+                'files': [],
+                'dirs': [],
+            }
+
+            # in older version, the binary used to be named 'EXAMPLE' rather than 'example'
+            if LooseVersion(self.version) < LooseVersion('1.0'):
+                custom_paths['files'].append('bin/EXAMPLE')
+            else:
+                custom_paths['files'].append('bin/example')
+
+            super(EB_Example, self).sanity_check_step(custom_paths=custom_paths)
+
+
+.. _implementing_easyblocks_module_only_compatibility:
+
+Compatibility with ``--extended-dry-run``/``-x`` and ``--module-only``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some special care must be taken to ensure that an easyblock is fully compatible with ``--extended-dry-run`` / ``-x``
+(see :ref:`extended_dry_run`) and ``--module-only`` (see :ref:`module_only`).
+
+For ``--extended-dry-run``/``-x``, this is already well covered at :ref:`extended_dry_run_guidelines_easyblocks_detect_dry_run`.
+
+For ``--module-only``, the main thing is to make sure that both the ``make_module_step``, including the ``make_module_*`` submethods,
```

Comment:
"the main thing" sounds pretty vague and informal. I wouldn't include it in a written tutorial, even if I use it when speaking the whole time.